File: analyze-bundle.js
const fs = require('fs');
const path = require('path');

const distPath = path.join(__dirname, 'dist', 'assets');

try {
  const files = fs.readdirSync(distPath);
  const jsFiles = [];
  const cssFiles = [];

  files.forEach(file => {
    const filePath = path.join(distPath, file);
    const stat = fs.statSync(filePath);
    const size = stat.size;

    if (file.endsWith('.js')) {
      jsFiles.push({ name: file, size });
    } else if (file.endsWith('.css')) {
      cssFiles.push({ name: file, size });
    }
  });

  // Sort by size descending
  jsFiles.sort((a, b) => b.size - a.size);
  cssFiles.sort((a, b) => b.size - a.size);

  // Calculate totals
  const totalJsSize = jsFiles.reduce((acc, f) => acc + f.size, 0);
  const totalCssSize = cssFiles.reduce((acc, f) => acc + f.size, 0);

  console.log('\n=== BUNDLE METRICS ===\n');
  console.log(`Total JS Size: ${totalJsSize} bytes (${(totalJsSize / 1024).toFixed(2)} KB)`);
  console.log(`Total CSS Size: ${totalCssSize} bytes (${(totalCssSize / 1024).toFixed(2)} KB)`);
  console.log(`\nJS Files Count: ${jsFiles.length}`);
  console.log(`CSS Files Count: ${cssFiles.length}`);

  console.log('\n=== TOP 5 LARGEST JS CHUNKS ===\n');
  jsFiles.slice(0, 5).forEach((f, i) => {
    console.log(`${i + 1}. ${f.name} - ${f.size} bytes (${(f.size / 1024).toFixed(2)} KB)`);
  });

  console.log('\n=== TOP 5 LARGEST CSS CHUNKS ===\n');
  cssFiles.slice(0, 5).forEach((f, i) => {
    console.log(`${i + 1}. ${f.name} - ${f.size} bytes (${(f.size / 1024).toFixed(2)} KB)`);
  });

  console.log('\n=== ALL JS FILES (sorted by size) ===\n');
  jsFiles.forEach(f => {
    console.log(`${f.name}|${f.size}`);
  });

  console.log('\n=== ALL CSS FILES (sorted by size) ===\n');
  cssFiles.forEach(f => {
    console.log(`${f.name}|${f.size}`);
  });

} catch (err) {
  console.error('Error:', err.message);
  process.exit(1);
}

File: analyze_bundle.py
#!/usr/bin/env python3

import os
import json
from pathlib import Path

dist_path = Path(__file__).parent / 'dist' / 'assets'

js_files = []
css_files = []

if dist_path.exists():
    for file in sorted(dist_path.iterdir()):
        if file.is_file():
            size = file.stat().st_size
            if file.suffix == '.js':
                js_files.append({'name': file.name, 'size': size})
            elif file.suffix == '.css':
                css_files.append({'name': file.name, 'size': size})

js_files.sort(key=lambda x: x['size'], reverse=True)
css_files.sort(key=lambda x: x['size'], reverse=True)

total_js = sum(f['size'] for f in js_files)
total_css = sum(f['size'] for f in css_files)

metrics = {
    'totalJsBytes': total_js,
    'totalCssBytes': total_css,
    'totalJsKb': round(total_js / 1024, 2),
    'totalCssKb': round(total_css / 1024, 2),
    'jsFileCount': len(js_files),
    'cssFileCount': len(css_files),
    'top5Js': [{'name': f['name'], 'bytes': f['size'], 'kb': round(f['size'] / 1024, 2)} for f in js_files[:5]],
    'top5Css': [{'name': f['name'], 'bytes': f['size'], 'kb': round(f['size'] / 1024, 2)} for f in css_files[:5]],
    'allJs': js_files,
    'allCss': css_files
}

print(json.dumps(metrics, indent=2))
